### PR TITLE
glimpse: init at 0.1.2

### PR DESCRIPTION
**Pull request by ashkitten**
###### Motivation for this change

requested in https://github.com/NixOS/nixpkgs/issues/77157

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/glimpse/default.nix [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+let
+  inherit (python2Packages) pygtk wrapPython python;
+in stdenv.mkDerivation rec {
+  pname = "glimpse";
+  version = "0.1.2";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchTarball {
```
Comment (jtojnar) [inline pkgs/applications/graphics/glimpse/default.nix]: We probably should not use `fetchTarball` directly but use `fetchurl` instead.

File: pkgs/applications/graphics/ufraw/default.nix [hunk 2]
```diff
@@ -60,7 +60,7 @@ stdenv.mkDerivation {
   ] ++ stdenv.lib.optional withGimpPlugin "--with-gimp";
 
   postInstall = stdenv.lib.optionalString withGimpPlugin ''
-    moveToOutput "lib/gimp" "$gimpPlugin"
+    moveToOutput ${gimp.targetPluginDir} "$gimpPlugin"
```
Comment (jtojnar) [inline pkgs/applications/graphics/ufraw/default.nix]: Could you move this to a separate commit?

Comment (ashkitten) [reply pkgs/applications/graphics/ufraw/default.nix]: sure

**Comment by jtojnar** [comment]
Did you decide to go with the plug-ins in the end?

**Comment by ashkitten** [comment]
i'm still on the edge about that, but i think we will be able to maintain compatibility with at least some of the plugins (g'mic and resynthesizer plugins are [slated to be included by default in official glimpse packages](https://github.com/glimpse-editor/Glimpse/issues/262)) and i think if we can't reasonably maintain compatibility with certain plugins in the future we can still drop them.

**Comment by ashkitten** [comment]
all checks passed!

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -19399,6 +19399,18 @@ in
...
   gimpPlugins = recurseIntoAttrs (callPackage ../applications/graphics/gimp/plugins {});
 
+  glimpse = callPackage ../applications/graphics/glimpse {
+    gegl = gegl_0_4;
+    lcms = lcms2;
+    inherit (darwin.apple_sdk.frameworks) AppKit Cocoa;
+  };
+
+  glimpse-with-plugins = callPackage ../applications/graphics/glimpse/wrapper.nix {
+    plugins = null; # All packaged plugins enabled, if not explicit plugin list supplied
```
Comment (gloaming) [inline pkgs/top-level/all-packages.nix]:
```suggestion
    plugins = null; # All packaged plugins enabled, if no list supplied
```

**Comment by gloaming** [comment]
The duplication here is a bit concerning. Would it not be better to override the gimp derivation where necessary?

**Comment by ashkitten** [comment]
i've decided to duplicate the derivation because while glimpse doesn't change much from its upstream at the moment, the plan is for it to evolve into an entirely separate project. it would be difficult to maintain if we packaged it as an override.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (ashkitten) [reply pkgs/top-level/all-packages.nix]: this is also an issue with the documentation for gimp. should i add another commit to fix it for them as well?

File: pkgs/applications/graphics/glimpse/default.nix [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  ];
+
+  # on Darwin,
+  # test-eevl.c:64:36: error: initializer element is not a compile-time constant
+  doCheck = !stdenv.isDarwin;
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "Fork of the GNU Image Manipulation Program";
```
Comment (grahamc) [inline pkgs/applications/graphics/glimpse/default.nix:171]:
```suggestion
    description = "Glimpse is an open source image editor based on the GNU Image Manipulation Program";
    longDescription = ''
      Glimpse is an open source image editor based on the GNU Image Manipulation Program. The goal is to experiment with new ideas and expand the use of free software.
    '';
```

maybe?

Comment (ashkitten) [reply pkgs/applications/graphics/glimpse/default.nix:171]: sure, i just took that description from the description on https://github.com/glimpse-editor/Glimpse

**Comment by grahamc** [comment]
building it I got a segfault, possibly related to this:

```
Glimpse-Warning: Bad interpreter referenced in interpreter file /nix/store/f30pn34ljwppl5bsrarszgdk4bly4p9b-glimpse-0.1.2/lib/glimpse/2.0/interpreters/pygimp.interp: python2

Glimpse-Warning: Bad binary format string in interpreter file /nix/store/f30pn34ljwppl5bsrarszgdk4bly4p9b-glimpse-0.1.2/lib/glimpse/2.0/interpreters/pygimp.interp

**Comment by ashkitten** [comment]
`> building it I got a segfault, possibly related to this:
> 
> ```
> Glimpse-Warning: Bad interpreter referenced in interpreter file /nix/store/f30pn34ljwppl5bsrarszgdk4bly4p9b-glimpse-0.1.2/lib/glimpse/2.0/interpreters/pygimp.interp: python2
> 
> Glimpse-Warning: Bad binary format string in interpreter file /nix/store/f30pn34ljwppl5bsrarszgdk4bly4p9b-glimpse-0.1.2/lib/glimpse/2.0/interpreters/pygimp.interp
> ```

a segfault while building? i've never gotten that before... also, iirc the `gimp` build has those same warnings

**Comment by grahamc** [comment]
oops, missed a word :)

**Comment by grahamc** [comment]
``````
Glimpse version 2.10.12
git-describe: f32663b334
C compiler:
gcc-9.3.0
using GEGL version 0.4.22 (compiled against version 0.4.22)
using GLib version 2.64.1 (compiled against version 2.64.1)
using GdkPixbuf version 2.40.0 (compiled against version 2.40.0)
using GTK+ version 2.24.32 (compiled against version 2.24.32)
using Pango version 1.44.7 (compiled against version 1.44.7)
using Fontconfig version 2.12.6 (compiled against version 2.12.6)
using Cairo version 1.16.0 (compiled against version 1.16.0)

```
> fatal error: Segmentation fault

Stack trace:
```
/nix/store/f30pn34ljwppl5bsrarszgdk4bly4p9b-glimpse-0.1.2/lib/libgimpbase-2.0.so.0(gimp_stack_trace_print+0x398)[0x7f532881f418]
./result/bin/glimpse[0x4ac6e0]
./result/bin/glimpse[0x4acb08]
./result/bin/glimpse[0x4ad199]
/nix/store/an6bdv4phxsz14q2sk57iscl2dc7bnj1-glibc-2.30/lib/libpthread.so.0(+0x12f70)[0x7f5327c22f70]
./result/bin/glimpse(gimp_param_spec_layer_id+0x64)[0x82be64]
./result/bin/glimpse(gimp_pdb_compat_param_spec+0x1c7)[0x746aa7]
./result/bin/glimpse(gimp_plug_in_handle_message+0x11a9)[0x7534e9]
./result/bin/glimpse(gimp_plug_in_manager_call_query+0x191)[0x761b41]
./result/bin/glimpse(gimp_plug_in_manager_restore+0x796)[0x759ab6]
./result/bin/glimpse[0x77776d]
/nix/store/idqwnnq6a1mzci2fi2rc26wqz9s0aa4a-glib-2.64.1/lib/libgobject-2.0.so.0(g_closure_invoke+0x1a2)[0x7f5327f9aec2]
/nix/store/idqwnnq6a1mzci2fi2rc26wqz9s0aa4a-glib-2.64.1/lib/libgobject-2.0.so.0(+0x2753e)[0x7f5327fad53e]
/nix/store/idqwnnq6a1mzci2fi2rc26wqz9s0aa4a-glib-2.64.1/lib/libgobject-2.0.so.0(g_signal_emit_valist+0xbdf)[0x7f5327fb8fef]
/nix/store/idqwnnq6a1mzci2fi2rc26wqz9s0aa4a-glib-2.64.1/lib/libgobject-2.0.so.0(g_signal_emit+0x8f)[0x7f5327fb997f]
./result/bin/glimpse(gimp_restore+0x102)[0x776cf2]
./result/bin/glimpse(app_run+0x49b)[0x4ac00b]
./result/bin/glimpse(main+0x37e)[0x4ab7be]
/nix/store/an6bdv4phxsz14q2sk57iscl2dc7bnj1-glibc-2.30/lib/libc.so.6(__libc_start_main+0xeb)[0x7f5327a74d8b]
./result/bin/glimpse(_start+0x2a)[0x4ab94a]

````

**Comment by ashkitten** [comment]
it works fine for me, even with those interpreter warnings you showed before

**Comment by glittershark** [comment]
I've been using this via @ashkitten's fork for a couple weeks now and it's been working great for me

**Comment by glittershark** [comment]
albeit very lightly, mostly for cropping and combining screenshots

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (gloaming) [reply pkgs/top-level/all-packages.nix:19409]: Ah, of course. Yes, though it should go in another PR I think.

**Comment by gloaming** [comment]
> i've decided to duplicate the derivation because while glimpse doesn't change much from its upstream at the moment, the plan is for it to evolve into an entirely separate project.

Hmm, this doesn't seem quite right - from the FAQ:

```
Is forking the project a duplication of effort?
No, because we are focusing specifically on the user experience and marketing, not on the underlying functionality of the software itself.
That is why we intend to periodically rebase on tagged versions of the GNU Image Manipulation Program, and then port our changes to it each time.
```

>  it would be difficult to maintain if we packaged it as an override.

This may still be true though.

**Comment by lourkeur** [comment]
I ran into the segfault too, but there's already a patch for it in upstream v0.1.2r2. There's no download for that release but it's easy enough to patch.

5c581ac9888a42c6213cbff7849a470f9872420f

**Comment by jtojnar** [comment]
The interpreter warning has been fixed in GIMP in #92228

**Comment by erictapen** [comment]
I'd like to help bring this forward.
Current PR is incompatible with fontconfig from `unstable`/`20.09`, so I rebased and bumped to Glimpse 0.2.0 in https://github.com/erictapen/nixpkgs/commit/82667f824b701551c29b1a5f6dd6f69a3d0a1cad.

So far this seems to run fine for me; No errors on startup, I was able to draw some basic stuff.

@ashkitten could you rebase and bump? I could also open a new PR if you'd like.

**Comment by ashkitten** [comment]
@erictapen i don't have the bandwidth for this rn so if you could take over that'd be fantastic

**Comment by erictapen** [comment]
Sure! Closing in favor of https://github.com/NixOS/nixpkgs/pull/103288.
